cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java

```diff
@@ -546,16 +428,18 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
         initialReconcile(context)
             .onComplete(context.succeeding(v -> context.verify(() -> {
                 originalPVCs.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
+
+                        //How to get the List of volumes here using the Pod
+
                         .map(StatefulSet::getSpec)
                         .map(StatefulSetSpec::getVolumeClaimTemplates)
                         .orElse(new ArrayList<>()));
-                originalVolumes.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
-                        .map(StatefulSet::getSpec)
-                        .map(StatefulSetSpec::getTemplate)
-                        .map(PodTemplateSpec::getSpec)
-                        .map(PodSpec::getVolumes)
-                        .orElse(new ArrayList<>()));
-                originalInitContainers.set(Optional.ofNullable(client.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
+
+                originalVolumes.set(Optional.ofNullable(supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME)).get())
```

Comment:
@scholzj This is the part where Iam majorly stuck at.

Reply:
The VolumeClaimTemplates are gone now. So there are only two things to check:
* The actual PVCs created by Strimzi
* The volumes of the pods in the StrimziPodSet (each pod should mount the right PVC)

I think that the `assertVolumes` method should be updated to cover the second part. The `PodSetUtils` class has utility methods for decoding the Pods from the StrimziPodSet resource to make it easy to check things. 

The `assertPVC` method should cover the first. It needs to be rewritten to check the PVCs and not the PersistentVolumeClaimTemplates.

TBH, I do not entirely understand how do the init containers fit here. IMHO they have nothing to do with persistent or ephemeral storage. But that should be easy to fix again by decoding the Pods from the StrimziPodSet and checking the pods.